spy/util.py

```diff
@@ -118,6 +120,28 @@ def highlight_C_maybe(code: str | bytes) -> str:
     return highlight(code, CLexer(), TerminalFormatter())
 
 
+def highlight_spy_maybe(code: str | bytes) -> str:
```

Comment:
good idea to have syntax highlighting also for spy code!
  1. `highlight_C_maybe` and `highlight_spy_maybe` have a lot of duplicated code. Maybe we can turn them into a single `highlight_src_maybe("C", ...)`
  2. the spy version uses CLexer which is wrong. It should use PythonLexer? Also, it might be worth using the lexer for Python 2.7, because it automatically highlights text in backticks, which we emit a lot after redshifting
  3. we should call this function also when we do a plain `--redshift`